cocos/ui/UIEditBox/UIEditBoxImpl-win32.cpp

```diff
@@ -416,15 +421,40 @@ LRESULT CALLBACK CWin32InputBox::DlgProc(HWND hDlg, UINT message, WPARAM wParam,
 					   {
 						   if (buttonId == definputbox_buttonids[i])
 						   {
-							   ::GetWindowTextA(
-								   _this->_hwndEditCtrl,
-								   _this->_param->szResult,
-								   _this->_param->nResultSize);
+							   std::u16string wstrResult;
+							   std::string utf8Result;
 
-							   std::string strUtf8 = AnsiToUtf8(_this->_param->szResult);
+							   int inputLength = ::GetWindowTextLengthW(_this->_hwndEditCtrl);
+							   wstrResult.resize(inputLength);
 
-							   memset(_this->_param->szResult, 0, _this->_param->nResultSize);
-							   strncpy(_this->_param->szResult, strUtf8.c_str(), _this->_param->nResultSize - 1);
+							   ::GetWindowTextW(
+								   _this->_hwndEditCtrl,
+								   (LPWSTR) const_cast<char16_t*>(wstrResult.c_str()),
+								   inputLength+1);
+
+							   bool conversionResult = cocos2d::StringUtils::UTF16ToUTF8(wstrResult, utf8Result);
+							   _this->_param->pstrResult->clear();
+							   if (conversionResult)
+							   {
+								   if((_this->_param->nMaxLength > 0) && (_this->_param->nMaxLength < (DWORD) cocos2d::StringUtils::getCharacterCountInUTF8String(utf8Result)))
```

Comment:
Could you break up this line? It's too long and don't use function call in compare operators.  Thanks.